src/aphroditeInterfaceFactory.js

```diff
@@ -62,6 +96,67 @@ export default ({ StyleSheet, css }/* aphrodite */) => ({
       inlineStyles,
     } = separateStyles(flattenedStyles);
 
+    const stylesWithDirection = [];
+    aphroditeStyles.forEach((stylesObj) => {
+      let definition = stylesObj._definition;
+      const rtlStyles = generateRTLStyles(definition);
+      if (rtlStyles) {
+        // This applies the rtlStyles whenever dir="rtl" is set. This is because
+        // the interface knows nothing about the current directional context.
+        definition = { ...definition, _rtl: rtlStyles };
+      }
+
+      stylesWithDirection.push({
+        ...stylesObj,
+        _definition: definition,
+      });
+    });
+
+    const result = {};
+    if (hasInlineStyles) {
+      const inlineRTLStyles = generateRTLStyles(inlineStyles);
+
+      if (inlineRTLStyles) {
+        // Because we know nothing about the current directional context, there
+        // is no way to determine whether or not the inline styles should be
+        // flipped. As a result, if there are inline styles to be flipped, we
+        // do so by converting them to classnames and providing styles in both
+        // an RTL and an LTR context.
+        // This may not work for all situations! For instance, when using inline
+        // styles to animate a component with react-motion or animated.js,
+        // converting to classes is likely to be way too slow. For those and
+        // other edge-cases, consumers should rely on the resolveNoRTL method
+        // instead.
+        const stylesDef = Object.assign(inlineStyles, { _rtl: inlineRTLStyles });
```

Comment:
Please ensure we're using the transform-replace-object-assign babel transform in this project.